### DISCUSSION TITLE
As you are using pegen, You may have a grammar file. It would be easier to add vscode language support for pixy if you have a grammar file.

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [versionprime/pixie#7] [Ideas]

**Comment by pradeeproark** [comment]
Hi,
Thanks for the suggestion. Yes I do want to add support for highlighting, code completion etc. for Pixy. For now, I set the file association for .pix files to Python as Pixy files are mostly python which works. For the embedded part in HTML  took a look at https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide, we should be able to reuse most of the work already done for python and html grammars if we can teach vscode the boundaries.

**Comment by itsmeadarsh2008** [comment]
Do you have any idea of how you would handle formatters and linters? A guy from Reddit asked me about this while I was building nutritious for Python. The nutritious was a project that was exactly similar or modern to Packed.

**Reply by pradeeproark** [reply]
Hi,
I checked out nutritious and Packed. Nice.  Re: formatters and linters, I haven't checked yet, but if we can get vscode to recognise the correct mix of languages as suggested in https://code.visualstudio.com/api/language-extensions/embedded-languages, I presume we would be able to get the benefits of reusing highlighters, linters and formatters for the individual languages.
